diff --git a/config/cci.m4 b/config/cci.m4
index 7afc60c..15ddff1 100644
--- a/config/cci.m4
+++ b/config/cci.m4
@@ -5,18 +5,7 @@ dnl Copyright © 2012 UT-Battelle, LLC.  All rights reserved.
 dnl Copyright © 2012 Oak Ridge National Labs.  All rights reserved.
 dnl
 
-# Define CCI configure command line arguments
-AC_DEFUN([CCI_DEFINE_ARGS],[
-    AC_ARG_ENABLE([picky],
-        [AC_HELP_STRING([--enable-picky],
-                        [Turn on maintainer-level compiler pickyness])])
-    AS_IF([test -d $srcdir/.hg -o -d $srcdir/.svn -o -d $srcdir/.git],
-          [CCI_DEVEL_BUILD=yes
-           AS_IF([test "$enable_picky" = ""],
-                 [AC_MSG_WARN([Developer build: enabling pickyness by default])
-                  enable_picky=yes])])
-])
-
+CCI_DEVEL_BUILD=no
 # Main CCI m4 macro
 #
 # Expects two or three paramters:
