diff --git a/src/plugins/ctp/verbs/ctp_verbs_api.c b/src/plugins/ctp/verbs/ctp_verbs_api.c
index e187ebd..016d97b 100644
--- a/src/plugins/ctp/verbs/ctp_verbs_api.c
+++ b/src/plugins/ctp/verbs/ctp_verbs_api.c
@@ -1063,7 +1063,7 @@ void rdma_destroy_ep(struct rdma_cm_id *id)
 	if (id->qp)
 		rdma_destroy_qp(id);
 
-	rdma_destroy_id(id);
+	//rdma_destroy_id(id);
 }
 #endif /* HAVE_RDMA_ADDRINFO */
 
@@ -2363,6 +2363,7 @@ static void verbs_destroy_conn(cci__ep_t *ep, cci__conn_t *conn)
 			      __func__, strerror(ret));
 		}
 	}
+    rdma_destroy_ep(vconn->id);
 
 	pthread_rwlock_wrlock(&vep->conn_tree_lock);
 	tdelete(&vconn->qp_num, &vep->conn_tree, verbs_compare_u32);
